pkg/handler/handler.go

```diff
@@ -425,21 +433,38 @@ func (m *Modifier) buildPodPatchConfig(pod *corev1.Pod) *podPatchConfig {
 	}
 
 	// Use the STS WebIdentity method if set
-	roleArn, audience, regionalSTS, tokenExpiration := m.Cache.Get(pod.Spec.ServiceAccountName, pod.Namespace)
-	if roleArn != "" {
-		tokenExpiration, containersToSkip := m.parsePodAnnotations(pod, tokenExpiration)
+	handler := make(chan any, 1)
+	response := m.Cache.GetOrNotify(pod.Spec.ServiceAccountName, pod.Namespace, handler)
+	key := pod.Namespace + "/" + pod.Spec.ServiceAccountName
+	if !response.FoundInSACache && !response.FoundInCMCache && m.saLookupGraceTime > 0 {
+		klog.Warningf("Service account %q not found in the cache. Waiting up to %s to be notified", key, m.saLookupGraceTime)
```

Comment:
nit: do we want to format `key` as a string, instead of a quoted string, on these warning messages?

```suggestion
		klog.Warningf("Service account %s not found in the cache. Waiting up to %s to be notified", key, m.saLookupGraceTime)
```

Not sure if it'll come up, but quoted strings will print some characters as escape sequences, which can make it harder to read.

Reply:
tbh. I have no opinion about that. Just changed it.